### PR TITLE
Improve macOS build docs

### PR DESCRIPTION
**Pull request by thislooksfun**
The existing macOS documentation is pretty good, but incomplete. This fills in some of the gaps.

### CODE REVIEW

File: Documentation/BuildInstructions.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ Notes:
 - Installing osxfuse for the first time requires enabling its system extension in System Preferences and then restarting your machine. The output from installing osxfuse with brew says this, but it's easy to miss.
 - bash is needed because the default version installed on macOS doesn't support globstar
 - If you install some commercial EXT2 macOS fs handler instead of osxfuse and fuse-ext2, you will need to `brew install e2fsprogs` to obtain `mke2fs` anyway.
-- As of 2020-08-06, you might need to tell the build system about your newer host compiler. Once you've built the toolchain, navigate to `Build/`, `rm -rf *`, then run `cmake .. -G Ninja -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10`, then continue with `ninja install` as usual.
+- As of 2020-08-06, you might need to tell the build system about your newer host compiler. Once you've built the toolchain, navigate to `Build/i686/`, `rm -rf *`, then run `cmake ../.. -G Ninja -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10`, then continue with `ninja install` as usual.
```
Comment (awesomekling) [inline Documentation/BuildInstructions.md:133]: Hmm, is this actually necessary? I never go into the `Build/i686/` directory?

File: Documentation/BuildInstructions.md [hunk 2]
```diff
@@ -111,7 +111,11 @@ Once you're in nix-shell, you should be able to follow the build directions.
 Make sure you have all the dependencies installed:
 
 ```console
-brew install coreutils qemu e2fsprogs m4 autoconf libtool automake bash gcc@10 ninja cmake
+# core
+brew install coreutils qemu bash gcc@10 ninja cmake
+
+# Fuse + ex2
```
Comment (awesomekling) [inline Documentation/BuildInstructions.md:117]: It's `ext2`, not `ex2` (also in commit message) :)

Comment (thislooksfun) [reply Documentation/BuildInstructions.md:117]: Whoops, fixed!

File: Documentation/BuildInstructions.md (hunk 1 above)
Comment (thislooksfun) [reply Documentation/BuildInstructions.md:133]: It's not strictly necessary, but it brings this portion of the docs in line with d0427b610c52a88f2e0a13f76fbe4c7c26769999

**Pull request merged**